cumulus_port/ingest/url_path_template.py

```diff
@@ -0,0 +1,70 @@
+# Ported from:
+# https://github.com/nasa/cumulus/blob/master/packages/ingest/src/url-path-template.js
+
+import re
+
+from cumulus_port._internal.jsonpath import get
+
+TEMPLATE_PATTERN = re.compile("{([^{}]+)}")
+EXPRESSION_PATTERN = re.compile(r"([^(]+)\(([^)]+)\)")
+
+
+def _get_single_value(data: dict, path: str):
+    values = get(data, path)
+
+    if not values:
+        raise Exception(f"Could not resolve path {repr(path)}")
+
+    if len(values) != 1:
+        raise ValueError(f"Path returned multiple values {repr(path)}")
+
+    return values[0]
+
+
+def evaluate_operation(name, args):
+    """evaluate the operation specified in template
+
+    :param name: the name of the operation
+    :param args: the args (in array) of the operation
+    :returns: str - the return value of the operation
+    """
+    raise NotImplementedError("evaluate_operation is not implemented yet")
```

Comment:
Should we add a TODO here? I search for TODOs, which makes it easy to figure out what still needs to get done.

Reply:
I would say no, because we aren't even using that feature. I figured if we ever did start using that and also needed it in some code that uses the cumulus port then we'd come back and implement it. In that sense we have no plans to ever implement it, so a TODO doesn't feel quite right. There's also a TON more cumulus code that hasn't been and probably won't ever be ported over.

Reply:
Sounds good 👍